pommerman/network/ion_server/match.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+"""IonServer Match handler - This contains functions responsible for playing matches (You shouldn't use this file directly)"""
+
+# Imports
+from multiprocessing import Pipe
+from .constants import SPCommands as spc
+from uuid import uuid4
+from os import listdir, makedirs
+from numpy import array
+from rapidjson import dumps, dump
+from gzip import compress
+from enum import Enum
+from numpy import ndarray, uint8, int64
+import pommerman
+
+
+# Functions
+def unique_uuid(dir):
+    """Generates a unique UUID and checks for collision"""
+    while (True):
+        uuid = str(uuid4())[:10]
+        try:
+            ls_dir = listdir(dir)
+        except FileNotFoundError:
+            makedirs(dir)
+            ls_dir = []
+        if not uuid + ".json" in ls_dir:
+            return uuid
+
+
+def resolve_classes(i):
+    """Resolves classes before JSON serialization"""
```

Comment:
ditto re Args, etc.
in this case, i have no idea what ```i``` could be and isinstance is called on it multiple times.

Reply:
The function is a nested loop that converts obs into JSONable types so `i` can represent multiple things.

Reply:
Then say that in the function! I am so confused why you are just answering me here.

Reply:
Okay...